base/src/main/java/uk/ac/ebi/atlas/testutils/SolrUtils.java

```diff
@@ -1,36 +1,65 @@
 package uk.ac.ebi.atlas.testutils;
 
+import org.apache.solr.client.solrj.SolrQuery;
 import org.springframework.stereotype.Component;
 import uk.ac.ebi.atlas.solr.cloud.SolrCloudCollectionProxyFactory;
 import uk.ac.ebi.atlas.solr.cloud.collections.AnalyticsCollectionProxy;
+import static uk.ac.ebi.atlas.solr.cloud.collections.AnalyticsCollectionProxy.AnalyticsSchemaField;
 import uk.ac.ebi.atlas.solr.cloud.search.SolrQueryBuilder;
-
 import java.util.concurrent.ThreadLocalRandom;
 
-import static uk.ac.ebi.atlas.solr.cloud.collections.AnalyticsCollectionProxy.BIOENTITY_IDENTIFIER;
+//import static uk.ac.ebi.atlas.solr.cloud.collections.AnalyticsCollectionProxy.BIOENTITY_IDENTIFIER;
 
 @Component
 public class SolrUtils {
     private static final int MAX_ROWS = 10000;
     private static final ThreadLocalRandom RNG = ThreadLocalRandom.current();
 
-    private SolrCloudCollectionProxyFactory solrCloudCollectionProxyFactory;
+    private AnalyticsCollectionProxy analyticsCollectionProxy;
+    // private BioentitiesCollectionProxy bioentitiesCollectionProxy;
 
     public SolrUtils(SolrCloudCollectionProxyFactory solrCloudCollectionProxyFactory) {
-        this.solrCloudCollectionProxyFactory = solrCloudCollectionProxyFactory;
+        analyticsCollectionProxy = solrCloudCollectionProxyFactory.create(AnalyticsCollectionProxy.class);
+        // bioentitiesCollectionProxy = solrCloudCollectionProxyFactory.create(BioentitiesCollectionProxy.class);
     }
 
-    public String fetchRandomGeneFromAnalytics() {
-        AnalyticsCollectionProxy analyticsCollectionProxy =
-                solrCloudCollectionProxyFactory.create(AnalyticsCollectionProxy.class);
+    public String fetchRandomGeneIdFromAnalytics() {
         SolrQueryBuilder<AnalyticsCollectionProxy> queryBuilder = new SolrQueryBuilder<>();
-        queryBuilder.setFieldList(BIOENTITY_IDENTIFIER);
-        queryBuilder.setRows(MAX_ROWS);
+        queryBuilder
+                .setFieldList(AnalyticsCollectionProxy.BIOENTITY_IDENTIFIER)
+                .setRows(MAX_ROWS);
+
         return analyticsCollectionProxy
                 .query(queryBuilder)
                 .getResults()
                 .get(RNG.nextInt(MAX_ROWS))
                 .getFieldValue("bioentity_identifier")
                 .toString();
     }
+
+    public String fetchRandomGeneIdFromAnalytics(AnalyticsSchemaField field, String term) {
```

Comment:
I'm not sure this method name or its parameters are descriptive enough about what is going on... The method is basically getting a random gene ID that satisfies a condition, but you wouldn't know what without looking at the contents.

Reply:
The fact that you pass a field and a value is a strong enough hint, IMO. Take, for instance, [ThreadLocalRandom.nextInt(int origin, int bound)](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ThreadLocalRandom.html#nextInt-int-int-); any random generator with arguments imposes certain conditions or restrictions on the returned value. I can add a comment if you want, but I think it’s quite straightforward.